src/wp-includes/webfonts-api/providers/class-wp-webfonts-google-provider.php

```diff
@@ -0,0 +1,157 @@
+ *
+ * @since 5.9.0
+ */
+class WP_Webfonts_Google_Provider extends WP_Webfonts_Provider {
+
+	/**
+	 * The provider's unique ID.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @var string
+	 */
+	protected $id = 'google';
+
+	/**
+	 * The provider's root URL.
+	 *
+	 * @since 5.9.0
+	 * @var string
+	 */
+	protected $root_url = 'https://fonts.googleapis.com/css2';
+
+	/**
+	 * Array of resources hints.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @var array
+	 */
+	protected $resource_hints = array(
+		'preconnect' => array(
+			array(
+				'href'        => 'https://fonts.gstatic.com',
+				'crossorigin' => 'anonymous',
+			),
+		),
+	);
+
+	/**
+	 * Get the CSS for a collection of fonts.
+	 *
+	 * @access public
+	 * @since  5.9.0
+	 * @return string
+	 */
+	public function get_css() {
+		$css  = '';
+		$urls = $this->build_collection_api_urls();
+
+		foreach ( $urls as $url ) {
+			$css .= $this->get_cached_remote_styles( 'google_fonts_' . md5( $url ), $url );
+		}
+
+		return $css;
+	}
+
+	/**
+	 * Build the API URL for a collection of fonts.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @return array Collection of font-family urls.
+	 */
+	protected function build_collection_api_urls() {
+		$font_families_urls = array();
+
+		// Iterate over each font-family group and build the API URL partial for that font-family.
+		foreach ( $this->organize_webfonts() as $font_display => $font_families ) {
+			$font_display_url_parts = array();
+			foreach ( $font_families as $font_family => $webfonts ) {
+				$normal_weights = array();
+				$italic_weights = array();
+				$url_part       = urlencode( $font_family );
+
+				// Build an array of font-weights for italics and default styles.
+				foreach ( $webfonts as $font ) {
+					if ( 'italic' === $font['font-style'] ) {
+						$italic_weights[] = $font['font-weight'];
+					} else {
+						$normal_weights[] = $font['font-weight'];
+					}
+				}
+
+				if ( empty( $italic_weights ) && ! empty( $normal_weights ) ) {
+					$url_part .= ':wght@' . implode( ';', $normal_weights );
+				} elseif ( ! empty( $italic_weights ) && empty( $normal_weights ) ) {
+					$url_part .= ':ital,wght@1,' . implode( ';', $normal_weights );
+				} elseif ( ! empty( $italic_weights ) && ! empty( $normal_weights ) ) {
+					$url_part .= ':ital,wght@0,' . implode( ';0,', $normal_weights ) . ';1,' . implode( ';1,', $italic_weights );
+				}
+
+				$font_display_url_parts[] = $url_part;
+			}
+
+			$font_families_urls[] = $this->root_url . '?family=' . implode( '&family=', $font_display_url_parts ) . '&display=' . $font_display;
+		}
+
+		return $font_families_urls;
+	}
+
+	/**
+	 * Organizes the webfonts by font-display.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @return array Sorted by font-display.
+	 */
+	private function organize_webfonts() {
+		$font_display_groups = array();
+
+		/*
+		 * Group by font-display.
+		 * Each font-display will need to be a separate request.
+		 */
+		foreach ( $this->webfonts as $webfont ) {
+			if ( ! isset( $font['font-display'] ) ) {
```

Comment:
I think this should be `$webfont`

```suggestion
			if ( ! isset( $webfont['font-display'] ) ) {
```